ZFSin/zfs/module/zfs/zfs_vnops_windows.c

```diff
@@ -1549,10 +1550,11 @@ NTSTATUS pnp_query_id(PDEVICE_OBJECT DeviceObject, PIRP Irp, PIO_STACK_LOCATION
 
 	zmo = (mount_t *)DeviceObject->DeviceExtension;
 
-	Irp->IoStatus.Information = (void *)ExAllocatePoolWithTag(PagedPool, zmo->bus_name.Length + sizeof(UNICODE_NULL), '!OIZ');
-	if (Irp->IoStatus.Information == NULL) return STATUS_NO_MEMORY;
+	Irp->IoStatus.Information = (ULONG_PTR) ExAllocatePoolWithTag(PagedPool, 2 * (zmo->bus_name.Length + 1), '!OIZ');
+	if (Irp->IoStatus.Information == (ULONG_PTR) NULL) return STATUS_NO_MEMORY;
 
-	RtlCopyMemory(Irp->IoStatus.Information, zmo->bus_name.Buffer, zmo->bus_name.Length);
+	RtlCopyMemory((void*)(Irp->IoStatus.Information), zmo->bus_name.Buffer, 2 * zmo->bus_name.Length);
+	((WCHAR*)(Irp->IoStatus.Information))[zmo->bus_name.Length] = L'\0';
```

Comment:
if you are addressing the array in WCHAR mode then you need to divide the length by 2.  Also, what was the rationale to terminate the string with NULL? is it required?

Reply:
This change is not required as the bus_name.Length was assumed to be in wchars not bytes. As we are allocating an extra wchar in ExAllocatePoolWithTag(), it prompted me to think we needed to NULL terminate the string.